dedupe/dedupe_test.go

```diff
@@ -26,18 +25,52 @@ func TestPacketDeduplicatorSeen(t *testing.T) {
 	}
 
 	// Wait for expiration
-	time.Sleep(expiresAfter + 100*time.Millisecond)
+	time.Sleep(expiresAfter + time.Millisecond)
 
 	// Test Seen with same packetID after expiration
 	if dedup.Seen(sender, packetID) {
 		t.Error("Expected the packet to not have been seen after expiration")
 	}
 }
 
+func TestDuplicatorProp(t *testing.T) {
+	if err := quick.Check(func(s, p uint32) bool {
+		dedup := NewDeduplicator(expiresAfter)
+		// Test Seen with new packetID
+		if dedup.Seen(s, p) {
+			t.Error("Expected the packet to not have been seen")
+			return false
+		}
+
+		// Test Seen with same packetID again
+		if !dedup.Seen(s, p) {
+			t.Error("Expected the packet to have been seen")
+			return false
+		}
+		return true
+	}, nil); err != nil {
+		t.Error(err)
+	}
+}
+
+func FuzzDup(f *testing.F) {
+	f.Fuzz(func(t *testing.T, s uint32, p uint32) {
```

Comment:
The fuzz tester actually allows property assertions that use coverage driven generation which is pretty great.  That's slightly pointless for this test case because it's obviously correct, but it's good in theory.